movethebytes/data-transfer/README.md

```diff
@@ -0,0 +1,275 @@
+Then call `testground run composition --file PATH/TO/COMPOSITION/FILE`.
+
+#### more notes on logging
+Testground will output any logs you add in the testplan and in your protocol, as long as you set up the logger in your testplan. 
+Testground will prefix those logs with the specific instance's sequence number and a hash id, so you will have context for which node is outputting what
+logging.
+
+## Basic example: TCP
+Let's walk through an example that removes all protocol complexity, and leaves only the testground complexity: it can be found at `test-plan/movethebytes/data-transfer/rust/bin/tcp.rs`.
+
+Let's have a test that spins up nodes that can connect over TCP, half of these nodes are going to be "server" nodes, the other half "client" nodes. When we run the test, we can run with only 2 nodes, one of which is a server, the other a client, but we will assume that we can have more than two instances.
+
+In general, the test is separated into 3 parts
+1) the main function
+2) the test function
+3) the "client" & "server" functions
+
+### The main function
+The job of the main function is to call the correct test function.
+In the tcp case, this just calls the `run_tcp_test`.
+You can technically run different tests from the same file by using
+different parameters, as you can see in the `sdk-rust` example: 
+`https://github.com/testground/sdk-rust/blob/master/examples/example.rs`
+
+### The test function
+This is where the testground `Client` gets built.
+
+The `Client` is how you have access to metadata about the instance itself, such
+as the `global_seq` number, which gets used in a few places, and how to
+coordinate with the other instances. This coordination happens "out of band" of
+whatever server/client configuration you write in the test itself. It's how you
+can pass around your instance's address & get the address of the other
+instances, as well as any generated Cids you need to exchange. You also use the
+client to sync the timing of things like:
+- ensuring instance A doesn't dial instance B until instance B has been created
+  & is waiting for a connection
+- ensuring instance C keeps running until instance D has been able to request
+  and verify some content from instance C
+
+The "run test" function not only creates the client, but also helps the test
+runner determine what "kind" of role this instance is going to have in the
+test. 
+
+In this particular case, we will only have 2 roles, a `server` and a `client`.
+In a p2p context `server` and `client` don't always make sense. For example, if
+you are testing a network's ability to connect to every other node in the
+network, you may only have one "role" that every node in that test takes on.
+You may have more than 2 roles!
+
+But in this case, and in the case of this data transfer plan (currently), we are
+only looking at point to point transfer between two nodes.
+
+We differentiate roles by using the global sequence number. In this case, nodes
```

Comment:
This is not ideal and should instead be based on the `TEST_GROUP_ID` param in testground. Unsure how exactly you access it with the rust tg client.
There are use cases where you have any number of groups. Ie multi version setups or passive nodes or varying config nodes etc.

Reply:
k, making a note in the doc and will update later